File: frontend/src/services/calibrationService.ts
import api from './api';
import { Calibration, CreateCalibrationPackageRequest, CandidateRanking, ApiResponse } from '@/types';

export const calibrationService = {
  createCalibration: async (gradeId: number): Promise<Calibration> => {
    const response = await api.post<ApiResponse<Calibration>>(`/calibrations?gradeId=${gradeId}`);
    return response.data.data;
  },

  createCalibrationPackage: async (request: CreateCalibrationPackageRequest): Promise<Calibration> => {
    const response = await api.post<ApiResponse<Calibration>>('/calibrations/package', request);
    return response.data.data;
  },

  updateStatus: async (id: number, status: string): Promise<Calibration> => {
    const response = await api.put<ApiResponse<Calibration>>(`/calibrations/${id}/status?status=${status}`);
    return response.data.data;
  },

  getCalibrationsByStatus: async (status: string): Promise<Calibration[]> => {
    const response = await api.get<ApiResponse<Calibration[]>>(`/calibrations/status/${status}`);
    return response.data.data;
  },

  getCalibrationDetails: async (id: number): Promise<Calibration> => {
    const response = await api.get<ApiResponse<Calibration>>(`/calibrations/${id}/details`);
    return response.data.data;
  },

  getCalibrationsByEvaluatorId: async (evaluatorId: number): Promise<Calibration[]> => {
    const response = await api.get<ApiResponse<Calibration[]>>(`/calibrations/evaluator/${evaluatorId}`);
    return response.data.data;
  },

  getCandidateRanking: async (id: number): Promise<CandidateRanking> => {
    const response = await api.get<ApiResponse<CandidateRanking>>(`/calibrations/${id}/ranking`);
    return response.data.data;
  },
};

File: backend/src/main/java/com/grading/service/CalibrationService.java
package com.grading.service;

import com.grading.dto.request.CreateCalibrationPackageRequest;
import com.grading.dto.response.CalibrationResponse;
import com.grading.dto.response.CandidateRankingResponse;
import com.grading.entity.Calibration;

import java.util.List;

public interface CalibrationService {
    Calibration createCalibration(Long gradeId, Long createdById);
    CalibrationResponse createCalibrationPackage(CreateCalibrationPackageRequest request, Long createdById);
    Calibration updateCalibrationStatus(Long id, String status);
    Calibration getCalibrationById(Long id);
    CalibrationResponse getCalibrationResponseById(Long id);
    List<Calibration> getCalibrationsByStatus(String status);
    List<CalibrationResponse> getCalibrationResponsesByStatus(String status);
    List<Calibration> getCalibrationsByGradeId(Long gradeId);
    List<CalibrationResponse> getCalibrationsByEvaluatorId(Long evaluatorId);
    CandidateRankingResponse getCandidateRanking(Long calibrationId);
}